File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/payment/application/integrationevent/PaymentCancelledIntegrationEventV1.java
package com.example.accountservice.payment.application.integrationevent;

/**
 * The Integration Event (public contract) that the Payment BC exposes to external BCs. Carries only
 * the minimal information Account needs to run a compensating credit (deposit).
 */
public record PaymentCancelledIntegrationEventV1(String paymentId, String accountId, long amount) {

    public static final String EVENT_TYPE = "payment.cancelled.v1";
}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/payment/application/integrationevent/PaymentCancelledIntegrationEventV1.kt
package com.example.accountservice.payment.application.integrationevent

import com.example.accountservice.outbox.IntegrationEventContract

/**
 * The Integration Event (public contract) that Payment BC exposes to an external BC (Account).
 * Carries only the minimal information Account needs to execute the compensating credit (deposit).
 */
data class PaymentCancelledIntegrationEventV1(
    val paymentId: String,
    val accountId: String,
    val amount: Long,
    val cancelledAt: String,
) : IntegrationEventContract {
    override val eventName: String get() = EVENT_NAME

    companion object {
        const val EVENT_NAME = "payment.cancelled.v1"
    }
}
